f/connectors/alerts/alerts_gcs.py

```diff
@@ -124,221 +131,251 @@ def _get_tif_rel_filepath(blob_name, local_file_path, territory_id):
-                continue
-
-            logger.info(f"Converting TIFF file to JPEG: {jpeg_file}")
-
-            # Ensure the file exists in the local directory
-            if os.path.exists(os.path.join(rel_filepath, file_name.split("/")[-1])):
-                try:
-                    with Image.open(
-                        os.path.join(rel_filepath, file_name.split("/")[-1])
-                    ) as img:
-                        # save the image at the same location in the datalake as its tiff
-                        img.save(os.path.join(rel_filepath, jpeg_file), "JPEG")
-                except Exception as e:
-                    logger.error(
-                        f"Tif image can not be opened, potentially empty: {str(e)}"
-                    )
-        elif file_name.endswith(".csv"):
-            if file_name != alerts_metadata_filename:
-                continue
-            file_path = os.path.join(path, file_name)
-            if os.path.exists(file_path):
-                df = pd.read_csv(file_path)
-                filtered_df = df[df["territory_id"] == territory_id]
-                # Hash the file content
-                hash_hex = sha256sum(file_path)
-                # Generate a UUID from the content hash
-                metadata_uuid = uuid.uuid5(uuid.NAMESPACE_DNS, hash_hex)
-
-                filtered_df["_geolocation"] = [[0.0, 0.0]] * len(filtered_df)
-                filtered_df["metadata_uuid"] = [str(metadata_uuid)] * len(filtered_df)
-                filtered_df["source"] = file_name
-                # this op is used for terras alerts only at this point
-                filtered_df["alert_source"] = "terras"
-                alerts_metadata = filtered_df.to_dict("records")
-        else:
-            logger.info(f"Skipping file: {file_name}")
-
-    return {"geojsons": geojsons, "alerts_metadata": alerts_metadata}
-
-
-def load_alerts_to_postgis(local_directory, processed):
-    outputs = {}
-    outputs["alerts_metadata"] = processed["alerts_metadata"]
-    outputs["geojsons"] = []
-
-    for file_name in processed["geojsons"]:
-        if os.path.exists(os.path.join(local_directory, file_name)):
+    # Convert CSV bytes to DataFrame
+    df = pd.read_csv(StringIO(alerts_metadata))
+
+    # Filter DataFrame based on territory_id
+    filtered_df = df[df["territory_id"] == territory_id].copy()
+
+    # Generate a unique UUID for each row
+    filtered_df["metadata_uuid"] = filtered_df.apply(_generate_uuid, axis=1)
+
+    # TODO: Currently, this script is only used for Terras alerts. Let's discuss a more sustainable approach with the alerts provider(s).
+    filtered_df.loc[:, "alert_source"] = "terras"
+
+    # Replace all NaN values with None
+    filtered_df = filtered_df.replace({nan: None})
+
+    # Convert DataFrame to list of dictionaries
+    prepared_alerts_metadata = filtered_df.to_dict("records")
+
+    logger.info("Successfully prepared alerts metadata.")
+
+    return prepared_alerts_metadata
```

Comment:
the old code also added "_geolocation"  and "source" columns.  Do we need that? 


(I am not sure what that was for)

Reply:
No. `_geolocation` was a superfluous (I assume accidental) addition. I thought I had clarified somewhere that `source` stored the name of the filename e.g. "alerts_history.csv" which I do not think is necessary to store.